gateway-modules/apim-ballerina-native-component/src/main/java/org/wso2/carbon/apimgt/ballerina/util/ListFiles.java

```diff
@@ -0,0 +1,103 @@
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.Attribute;
+import org.ballerinalang.natives.annotations.BallerinaAnnotation;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+
+
+/**
+ * Ballerina function to return a sorted string array of json file names in a given folder.
+ * <br>
+ * Returns an empty array if the folder does not exist or when there are no json files inside
+ * org.wso2.carbon.apimgt.ballerina.util:listFiles
+ */
+@BallerinaFunction(
+        packageName = "org.wso2.carbon.apimgt.ballerina.util",
+        functionName = "listJSONFiles",
+        args = {@Argument(name = "folderPath", type = TypeEnum.STRING)},
+        returnType = {@ReturnType(type = TypeEnum.ARRAY, elementType = TypeEnum.STRING)},
+        isPublic = true
+)
+@BallerinaAnnotation(annotationName = "Description", attributes = {@Attribute(name = "value",
+        value = "Returns an array of json file names")})
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "folderPath",
+        value = "The path of the folder")})
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "list",
+        value = "json file array")})
+
+public class ListFiles extends AbstractNativeFunction  {
+    //private static final Log log = LogFactory.getLog(ListFiles.class);
+
+    @Override
+    public BValue[] execute(Context context) {
+        String directoryPath = getStringArgument(context, 0);
+        File directory = new File(directoryPath);
+        File[] fList = directory.listFiles();
+
+        List<String> list = new ArrayList<String>();
+        if (fList != null) {
+            //log.info("not null");
+            for (File file : fList) {
+                if (file.isFile() && file.getName().toLowerCase(Locale.ENGLISH).endsWith(".json")) {
+                    list.add(file.getName());
+                }
+            }
+            Collections.sort(list, Comparator.naturalOrder());
+            BStringArray balArray = new BStringArray();
+            int i = 0;
+            while (i < list.size()) {
+                balArray.add(i, list.get(i));
+                i = i + 1;
+            }
+
+            return getBValues(balArray);
+        } else {
+            return getBValues(new BStringArray());
+        }
+
+
+//        if (fList != null) {
+//            BStringArray balArray = new BStringArray();
+//            balArray.add(0, "a");
+//            balArray.add(1, "b");
+//            return getBValues(balArray);
+//        } else {
+//            BStringArray baArray = new BStringArray();
+//            baArray.add(0, "A");
+//            baArray.add(1, "B");
+//            return getBValues(baArray);
+//        }
```

Comment:
Is there any reason to comment these? Can we clean up here if they are not needed?